### PR TITLE
fix(clerk-js): Deprecate `afterSignOutUrl` from UserButton

### PR DESCRIPTION
**Pull request by panteliselef**
## Description

Fixes: DASH-183

This PR address the following issues:
- In a single session app, user opens user profile through the user button, the developer has set `afterSignOutUrl` but when user deletes their account, the provided `afterSignOutUrl` is not respected.
- In a multi session app, user goes to user profile, developer has set `afterMultiSessionSingleSignOutUrl` in user button but the user when they delete one of their accounts is not redirected to the appropriate url defined as prop.

The main issue is that we don't propagate `afterSignOutUrl` and `afterMultiSessionSingleSignOutUrl` from the UserButton to UserProfile.

This PR addresses that by positioning those props at the root lever (ClerkProvider or Clerk.load). We considered simply introducing those props to UserProfile, but the naming would not be ideal. We also considered renaming them, but at that point  you want the components to always be in sync, so we positioned them at the root.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx
Comment (BRKalow) [inline packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx:46]: 💭  I wouldn't necessarily expect that this method would need to be aware of multi-session. I understand why it was done this way as part of this PR, but it feels like a better internal abstraction might be to consolidate the multi-session logic into `navigateAfterSignOut()`.

File: packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx
Comment (BRKalow) [inline packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx:42]: Do we need to do this investigation before merge?

Comment (panteliselef) [reply packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx:42]: I was waiting for #3518 to be merged, and i will check again, but simply calling setActive would not refresh the router.

File: packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx
Comment (panteliselef) [reply packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx:46]: I think creating an abstraction here would make our lifer harder, we have `useMultisessionActions` which needs to know the difference between single session sign out and multisession sign out. We can refactor but i don't see a clear benefit here.

File: packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx
Comment (panteliselef) [reply packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx:42]: So i investigated further, even with #3518  the issue persisted. I've identified this as a FAPI issue, because after `user.delete()` client_uat comes back with a valid timestamp instead of being `0`. This will need a separate ticket to be address and should not be part of this PR

File: packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx
Comment (BRKalow) [reply packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx:46]: Benefit would be not having to spread some of this logic across many different methods. That's fine to leave it for now, let's jut keep an eye on these code paths and we can abstract if it feels like multi-session logic is too difficult to manage.

File: .changeset/selfish-ladybugs-smile.md
Comment (BRKalow) [inline .changeset/selfish-ladybugs-smile.md:6]: Can you specify what actions a user needs to take here w.r.t. the deprecation? What do they also need to change in their code?

**Review by BRKalow — APPROVED** [review]
Left a comment on the changelog, but otherwise looks good.

**Review by anagstef — APPROVED** [review]
LGTM 💯

File: .changeset/selfish-ladybugs-smile.md
Comment (panteliselef) [reply .changeset/selfish-ladybugs-smile.md]: makes sense, i updated it

**Pull request merged**
